tests/GraphCompareTest.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+import os
+import json
+import string 
+import random
+from .constants.TestPayload import *
+from .constants.TestEnvironment import *
+
+
+from app import app
+
+CURRENT_DIRECTORY = os.path.dirname(__file__)
+
+
+
+
+class GraphCompareTests(unittest.TestCase):
+
+    def setUp(self):
+        self.app = app.test_client()
+
+    def test_invalid_input_provided(self):
+
+        payload = dict(graphs="", dataset="RELIABLE")
+
+        headers = {}
+
+        response = self.app.post(
+            STRING_URL,
+            content_type = 'multipart/form-data',
+            headers=headers,
+            data=payload
+        )
+
+        self.assertEqual(400, response.status_code)
+        self.assertEqual('error' in STRING_OUTPUT_INVALID_INPUT, 'error' in response.json)
+
+    def test_invalid_dataset_provided(self):
+
+        letters = string.ascii_lowercase
+        dataset_name = ''.join(random.choice(letters) for i in range(10))
+        payload = dict(graphs=STRING_INPUT, dataset=dataset_name)
+
+        print(dataset_name)
+        headers = {}
+
+        response = self.app.post(
+            STRING_URL,
+            content_type = 'multipart/form-data',
+            headers=headers,
+            data=payload
+        )
+
+        self.assertEqual(200, response.status_code)
```

Comment:
invalid dataset still passes?

Reply:
if this is a test case that's testing for a valid case, you should rename it to something more appropriate

Reply:
this specific case is supposed to pass given the wrong dataset, the tool always defaults to a functional input for that parameter